mlrun/api/db/sqldb/models.py

```diff
@@ -225,6 +227,60 @@ def spec(self):
         def spec(self, value):
             self._spec = pickle.dumps(value)
 
+    class Feature(Base):
+        __tablename__ = "features"
+        id = Column(Integer, primary_key=True)
+        feature_set_id = Column(Integer, ForeignKey("feature_sets.id"))
+
+        name = Column(String)
+        value_type = Column(String)
+        # Either 'entity' or 'feature', according to the type of the feature
+        type = Column(String)
```

Comment:
There is a double-join between the feature_sets table and this table - both feature_sets.features and feature_sets.entities are pointing to this same table, and the difference between them is that their type is different (see the relationship definition in feature_sets to view the difference). 
value_type is simply the type of the feature (str/int/float etc).

Reply:
My point is that I don't fully understand what is feature and what is an entity, if they represent a different resource (e.g. feature set has list of features, and list of entities) I don't see a reason for them to be in the same table, they may be similar now, but tomorrow we'll want to add a field only to entities, if they represent different things, I suggest to have 2 different tables.
If they really represent 2 types of feature, it doesn't make sense that one of the types is... feature.

Reply:
In a sense, entity is a feature that is used for different purposes (used as axis for aggregation, and as sort of a "primary key" in a feature-set). To my understanding they are both instances of the same object, and that's why I kept them both in the same table. More so, a feature in feature-set 1 can serve as an entity in feature-set 2. It won't be the same DB-wise, since we're inserting new values to the tables per feature-set object, but they mean the same thing.
In short, I prefer to keep them as instances of the same object and hence as rows of the same table. If we come to a situation where we add fields to entities and not to features, then we need to question the entire logic behind this, but I think it's sound. At that point in time we may also need to restructure the DB anyway, given that the logic is different.

Reply:
After talking to @yaronha - I have split them into two separate tables.